lib/clientstate.ts

```diff
@@ -49,18 +49,20 @@ export class ClientStateCompilerOptions {
 }
 
 export class ClientStateCompiler {
-    constructor(jsondata) {
-        this._internalid = undefined;
-
+    _internalid: any = undefined;
+    id = '';
+    options = '';
+    filters: ClientStateCompilerOptions;
+    libs: any[] = [];
+    specialoutputs: any[] = [];
+    tools: any[] = [];
```

Comment:
If `jsondata` is provided `this.fromJsonData` will overwrite all these anyway so there should be no harm pulling initializations from the else branch up here